### PR TITLE
Added interpolation methods

### PR DESCRIPTION
**Pull request by og3b**
Added linear interpolation and look back interpolation.

Also added 'Chronological' which allows comparing the same type
 of DateRange.

### CODE REVIEW

**Comment by og3b** [comment]
@thoozaho could you review please.

curious to see what you think of using type classes for something like Numberlike. it means we can use the same code for interpolating doubles and quantities.

same for chronological. it means if you wanted to do something like add chronological to a joda time class you could just add the implicit object definition. like pimping but it doesn't create an instance (pimping does by creating a wrapper)

File: daterange/src/evcel/daterange/DateRange.scala [hunk 1]
```diff
@@ -52,4 +52,8 @@ object DateRange{
   val SIMPLE = "Simple"
 }
 
-case class SimpleDateRange(firstDay: Day, lastDay: Day) extends DateRange
\ No newline at end of file
+case class SimpleDateRange(firstDay: Day, lastDay: Day) extends DateRange
+
+trait Chronological[T <: DateRange] {
```
Comment (thoozaho) [inline daterange/src/evcel/daterange/DateRange.scala]: Class level comment?

**Comment by thoozaho** [comment]
Looks good - not sure I get all the implicit stuff, but I think that's me. Maybe give me a quick walkthrough tomorrow?

**Comment by og3b** [comment]
@thoozaho I've updated the PR as requested. Also change DateRange so that it's not ordered, but Day and Month are (as discussed). I'll merge tonight unless you say not to.

**Pull request merged**